### PR TITLE
Arm thumb/new ops

### PR DESCRIPTION
**Pull request by vdf-git**
new implementations of ops starting with a, b and c

### CODE REVIEW

**Comment by radare** [comment]
sames goes here, dont ever merge in your branches

File: libr/asm/arch/arm/armass.c [hunk 1]
```diff
@@ -152,42 +152,67 @@ static ArmOp ops[] = {
...
+static const ut32 X_BIT = 0x200;
+static const ut32 TWO_BIT = 0x400;
+static const ut32 IE_BIT = 0x800;
+static const ut32 ID_BIT = 0x1000;
+
+st8 iflag(char *input) {
+	st8 res = 0;
+	ut8 i;
+	r_str_case (input, false);
+
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: empty line

File: libr/asm/arch/arm/armass.c [hunk 2]
```diff
@@ -152,42 +152,67 @@ static ArmOp ops[] = {
...
+	
+	for (i = 0; i < strlen(input); i++) {
+		switch (input[i]) {
+		case 'a': {
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: unnecessary braces {}

File: libr/asm/arch/arm/armass.c [hunk 3]
```diff
@@ -200,27 +225,50 @@ ut32 opmask(char *input, char *opcode) {
 			// default is nv (no value)
 			res |= 15 << 2;
 		}
-		if (input[0]==0) {
-			return res;
+		if (!strncmp (input, "l", 1)) {
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: what about if (*input == 'l') instead of calling strncmp for 1 char?

File: libr/asm/arch/arm/armass.c [hunk 4]
```diff
@@ -200,27 +225,50 @@ ut32 opmask(char *input, char *opcode) {
...
+			res |= 1 << 8;
+			input++;
+		}
+		if (!strncmp (input, "x", 1)) {
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: same here

File: libr/asm/arch/arm/armass.c [hunk 5]
```diff
@@ -200,27 +225,50 @@ ut32 opmask(char *input, char *opcode) {
...
+			res |= 1 << 9;
+			input++;
+		}
+		for (i = 0; conds[i]; i++) {
+			if (!strncmp (input, conds[i], strlen (conds[i]))) {
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: you can use r_str_startswith

File: libr/asm/arch/arm/armass.c [hunk 6]
```diff
@@ -200,27 +225,50 @@ ut32 opmask(char *input, char *opcode) {
...
+				res &= i << 2; // NB: cond should be 1f here, otherwise the op is undefined
+				input+= strlen (conds[i]);
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: missing space before +

File: libr/asm/arch/arm/armass.c [hunk 7]
```diff
@@ -229,7 +277,7 @@ static int getnum(const char *str) {
 	while (*str == '$' || *str == '#') {
 		str++;
 	}
-	val = strtod (str, &endptr);
+	val = strtol (str, &endptr,0);
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: missing space after ,

File: libr/asm/arch/arm/armass.c [hunk 8]
```diff
@@ -699,11 +810,13 @@ static int thumb_assemble(ArmOpcode *ao, ut64 off, const char *str) {
 		ao->o = 0x40bf;
 		return 2;
 	} else
+#if 0
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: uh, 2hy kill3e bkp5, qne, ... ?

File: libr/asm/arch/arm/armass.c [hunk 9]
```diff
@@ -1185,7 +1304,704 @@ static int thumb_assemble(ArmOpcode *ao, ut64 off, const char *str) {
...
+		        }
+			break;
+		default:
+			return -1;
+		}
+	} else
+	if (( m = opmask (ao->op, "b") )) {
+		ut64 argt = thumb_selector (ao->a);
+		switch ( ((L_BIT | X_BIT) & m)) {
+		case (0x300): {
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: remove parenthesis

File: libr/asm/arch/arm/armass.c [hunk 10]
```diff
@@ -1185,7 +1304,704 @@ static int thumb_assemble(ArmOpcode *ao, ut64 off, const char *str) {
...
+		        }
+			break;
+		default:
+			return -1;
+		}
+	} else
+	if (( m = opmask (ao->op, "bfi") )) {
+		ut64 argt = thumb_selector (ao->a);
+		switch (argt) {
+		case 0x2211: {
```
Comment (radare) [inline libr/asm/arch/arm/armass.c]: what about using constants or enum instead of magic numbers?

**Comment by vdf-git** [comment]
changes put into pr #9728. Closing...

**Pull request closed without merging**